src/components/site/tableOfContents/index.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import reactn from 'reactn';
+import { Link, graphql, StaticQuery } from 'gatsby';
+import PropTypes from 'prop-types';
+import { Drawer } from 'react-md';
+import styles from './table-of-contents.module.scss';
+
+// import Footer from '../components/site/footer';
+const TO_PREFIX = '/';
+const routes = [
+  {
+    label: 'home',
+    to: `${TO_PREFIX}`,
+    icon: 'home',
+    exact: 'true',
+    primaryText: 'Home',
+    active: false,
+  },
+  {
+    label: 'style-guide',
+    to: `${TO_PREFIX}StyleGuide`,
+    icon: 'style',
+    primaryText: 'Style Guide',
+    active: false,
+  },
+  { divider: true },
+  {
+    primaryText: 'Table of Contents',
+    subheader: true,
+  },
+];
+
+@reactn
+class TableOfContents extends React.PureComponent {
+  // constructor(props) {
+  //   super(props);
+  //
+  // }
+
+  componentDidMount() {
+    const { navLinks } = this.props;
+    const links = routes.concat(navLinks);
+    this.navLinks = links.map(link => {
+      if (link.divider || link.subheader) return link;
+
+      if (link.to) {
+        link.component = Link;
+      }
+
+      if (link.slug) {
+        link.component = Link;
+        link.label = link.title;
+        link.to = `${TO_PREFIX}${link.slug}`;
+        link.primaryText = link.title;
+        link.active = false;
+      }
+      return link;
```

Comment:
this is totally fine, just wanted to point out another spread operator possibility:
```
return ({
   ...link,
   component: Link,
   label: link.title,
   to: `${TO_PREFIX}${link.slug}`,
   primaryText: link.title,
   active: new condition and stuff,
})
```
I haven't incorporated your `link.slug` check, but I'm not totally sure if that's necessary here.  I mean if it's not present it's the map function isn't like gonna throw out that item so you'd still end up passing stuff to the `navItems` prop that potentially won't render correctly so....yea, I don't know, I like the safe-guarding thinking, but I'm not sure the check is necessary at this juncture or doing the job you want it to.

Reply:
I get what you're saying here. So, here's what I did:

`componentDidMount() {
    const { navLinks } = this.props;
    this.navLinks = [
      ...this.routes,
      ...navLinks.map(link => {
        if (link.divider || link.subheader) return link;

        if (link.slug) {
          link.component = Link;
          link.label = link.title;
          link.to = `${this.TO_PREFIX}${link.slug}`;
          link.primaryText = link.title;
          link.active = false;
        }
        return {
          ...link,
          component: Link,
          label: link.title,
          to: `${this.TO_PREFIX}${link.slug}`,
          primaryText: link.title,
          active: false,
        };
      }),
    ];
  }`